slack-check-runs-notify/src/main.ts

```diff
@@ -0,0 +1,78 @@
+import {Octokit} from '@octokit/rest'
+import * as github from '@actions/github'
+import * as core from '@actions/core'
+
+const bent = require('bent')
+
+const [owner, repo] = process.env.GITHUB_REPOSITORY!.split('/', 2)
+const GITHUB_RUN_ID = Number.parseInt(process.env.GITHUB_RUN_ID!)
+
+const webhookToken = core.getInput('WEBHOOK_TOKEN')
+const commitMessage = core.getInput('COMMIT_MESSAGE')
+const GITHUB_TOKEN = core.getInput('GITHUB_TOKEN')
+
+const octokit: Octokit = (new github.GitHub(GITHUB_TOKEN) as any) as Octokit
+
+const runLink = `https://github.com/${owner}/${repo}/actions/runs/${runId}`
+const commit =
+  commitMessage!.trim().length > 0
+    ? `\n*${commitMessage!.trim().split('\n')[0]}*\n`
+    : ''
+
+const deploymentTestFail = {
+  text: `❌ A test check failed preventing deployment on master ${commit}<${runLink}|See github action>`
```

Comment:
We could even leave that as an input so we can use it for more than just master deployments on other repos, and maybe have a generic sentence like `A check failed for commit: <see github action>` if there's nothing set in the input